internal/state.go

```diff
@@ -74,13 +76,12 @@ func PrepareEngineState(ctx context.Context, engine string, appID string, ttl st
 	if appID == "" {
 		fmt.Println("No app specified, using session app.")
 	}
-	go func() {
-		for x := range global.SessionMessageChannel() {
-			if x.Topic != "OnConnected" {
-				fmt.Println(x.Topic, string(x.Content))
-			}
-		}
-	}()
+	sessionMessages := global.SessionMessageChannel()
+	err := waitForOnConnectedMessage(sessionMessages)
+	if err != nil {
+		FatalError("Failed to connect to engine with error message: ", err)
+	}
+	go printSessionMessagesIfInVerboseMode(sessionMessages)
```

Comment:
We are "looping" over the same channel twice. Doesn't this mean that the second loop will never receive the items that the first loop consumed?

Reply:
True, will fix